inception/decoder.go

```diff
@@ -136,6 +136,13 @@ func handleFieldAddr(ic *Inception, name string, takeAddr bool, typ reflect.Type
 				Ptr:             reflect.Ptr,
 				UseReflectToSet: useReflectToSet,
 			})
+		} else if typ.Elem().Kind() == reflect.Struct && typ.Name() != "" {
+			out += tplStr(decodeTpl["handleArray"], handleArray{
+				IC:   ic,
+				Name: name,
+				Typ:  typ,
+				Ptr:  reflect.Ptr,
+			})
 		} else if (typ.Elem().Kind() == reflect.Struct || typ.Elem().Kind() == reflect.Map) ||
```

Comment:
It would be ideal to kinda cleanup this `else if` branch, like why specifically is this fallback, but the above case is ok?  Basically, can we restructure the if statements so its clear why we fallback for a specific type?